consensus/src/main/kotlin/maru/consensus/ValidatorProvider.kt

```diff
@@ -0,0 +1,39 @@
+/*
+   Copyright 2025 Consensys Software Inc.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package maru.consensus
+
+import kotlin.collections.setOf
+import maru.core.Validator
+
+/**
+ * Provides access to the set of validators for a given block.
+ */
+interface ValidatorProvider {
+  fun getValidatorsAfterBlock(blockNumber: ULong): Set<Validator>
```

Comment:
This can have a default implementation ?
```suggestion
  fun getValidatorsAfterBlock(blockNumber: ULong): Set<Validator> = getValidatorsForBlock(blockNumber + 1uL)
```